### ISSUE TITLE
[App]: PAGE_FAULT_IN_NONPAGED_AREA Win10 blue screen of death error with pytorch lightning

### ISSUE THREAD

**Issue by maxencewynen** [wandb/client#3819]
### Current Behavior

Hi!

Everytime I try to run pytorch lightning with a wandb logger, my (windows 10) computer crashes and displays a PAGE_FAULT_IN_NONPAGED_AREA error (see attached image).
![Image d’iOS](https://user-images.githubusercontent.com/17831504/174565932-4948bdd6-9acf-4335-b045-95a11b9efa9f.jpg)

### Expected Behavior

No crash

### Steps To Reproduce

This is the basic file I run, heavily inspired by [this]( https://wandb.ai/manan-goel/MNIST/reports/How-to-Integrate-PyTorch-Lightning-with-Weights-Biases--VmlldzoxNjg1ODQ1)

```
from torchvision.datasets import MNIST
from torchvision import transforms
from torch.utils.data import DataLoader, random_split
import torch
from torch.nn import Linear, CrossEntropyLoss, functional as F
from torch.optim import Adam
from torchmetrics.functional import accuracy
from pytorch_lightning import LightningModule
from pytorch_lightning.callbacks import ModelCheckpoint
from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning import Trainer
from pytorch_lightning.callbacks import Callback
import wandb

class LogPredictionsCallback(Callback):

    def on_validation_batch_end(
            self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
        """Called when the validation batch ends."""

        # `outputs` comes from `LightningModule.validation_step`
        # which corresponds to our model predictions in this case

        # Let's log 20 sample image predictions from first batch
        if batch_idx == 0:
            n = 20
            x, y = batch
            images = [img for img in x[:n]]
            captions = [f'Ground Truth: {y_i} - Prediction: {y_pred}' for y_i, y_pred in zip(y[:n], outputs[:n])]

            # Option 1: log images with `WandbLogger.log_image`
            wandb_logger.log_image(key='sample_images', images=images, caption=captions)

            # Option 2: log predictions as a Table
            columns = ['image', 'ground truth', 'prediction']
            data = [[wandb.Image(x_i), y_i, y_pred] for x_i, y_i, y_pred in list(zip(x[:n], y[:n], outputs[:n]))]
            wandb_logger.log_table(key='sample_table', columns=columns, data=data)



class MNIST_LitModule(LightningModule):

    def __init__(self, n_classes=10, n_layer_1=128, n_layer_2=256, lr=1e-3):
        '''method used to define our model parameters'''
        super().__init__()

        # mnist images are (1, 28, 28) (channels, width, height)
        self.layer_1 = Linear(28 * 28, n_layer_1)
        self.layer_2 = Linear(n_layer_1, n_layer_2)
        self.layer_3 = Linear(n_layer_2, n_classes)

        # loss
        self.loss = CrossEntropyLoss()

        # optimizer parameters
        self.lr = lr

        # save hyper-parameters to self.hparams (auto-logged by W&B)
        self.save_hyperparameters()

    def forward(self, x):
        '''method used for inference input -> output'''

        batch_size, channels, width, height = x.size()

        # (b, 1, 28, 28) -> (b, 1*28*28)
        x = x.view(batch_size, -1)

        # let's do 3 x (linear + relu)
        x = self.layer_1(x)
        x = F.relu(x)
        x = self.layer_2(x)
        x = F.relu(x)
        x = self.layer_3(x)

        return x

    def training_step(self, batch, batch_idx):
        '''needs to return a loss from a single batch'''
        _, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('train_loss', loss)
        self.log('train_accuracy', acc)

        return loss

    def validation_step(self, batch, batch_idx):
        '''used for logging metrics'''
        preds, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('val_loss', loss)
        self.log('val_accuracy', acc)

        # Let's return preds to use it in a custom callback
        return preds

    def test_step(self, batch, batch_idx):
        '''used for logging metrics'''
        _, loss, acc = self._get_preds_loss_accuracy(batch)

        # Log loss and metric
        self.log('test_loss', loss)
        self.log('test_accuracy', acc)

    def configure_optimizers(self):
        '''defines model optimizer'''
        return Adam(self.parameters(), lr=self.lr)

    def _get_preds_loss_accuracy(self, batch):
        '''convenience function since train/valid/test steps are similar'''
        x, y = batch
        logits = self(x)
        preds = torch.argmax(logits, dim=1)
        loss = self.loss(logits, y)
        acc = accuracy(preds, y)
        return preds, loss, acc




if __name__ == '__main__':
    transform = transforms.Compose([
                transforms.ToTensor(),
                transforms.Normalize((0.1307,), (0.3081,))])

    dataset = MNIST(root="./MNIST", download=True, transform=transform)
    training_set, validation_set = random_split(dataset, [55000, 5000])

    training_loader = DataLoader(training_set, batch_size=64, shuffle=True)
    validation_loader = DataLoader(validation_set, batch_size=64)

    model = MNIST_LitModule(n_layer_1=128, n_layer_2=128)

    checkpoint_callback = ModelCheckpoint(monitor='val_accuracy', mode='max')

    wandb_logger = WandbLogger(project='MNIST',  # group runs in "MNIST" project
                               log_model='all')  # log all new checkpoints during training

    log_predictions_callback = LogPredictionsCallback()

    trainer = Trainer(
        logger=wandb_logger,  # W&B integration
        callbacks=[log_predictions_callback,  # logging of sample predictions
                   checkpoint_callback],  # our model checkpoint callback
        max_epochs=5)  # number of epochs

    trainer.fit(model, training_loader, validation_loader)

    wandb.finish()
```

### Environment

OS: Windows 10 21H2

Browsers: Chrome

Version: 
wandb=0.12.18
torch=1.11.0
pytorch_lightning=1.6.4

**Comment by MBakirWB** [comment]
Hi @maxencewynen 

I’ll be glad to look into this for you. We will need some additional information before we can proceed to troubleshoot. Please provide me with the following:

- Is your code running any containers
- Debug logs (internal-debug.log and debug.log) found in your wandb run directory
- Any additional details would be great for us to isolate the issue.

**Comment by MBakirWB** [comment]
@maxencewynen, 

Thank you for the info. Taking it a step back we have to isolate the issue to either being a wandb or windows related error. 

Starting with wandb, 

1. Assuming that you followed these[ installation instructions](https://pytorch.org/get-started/locally/) for pytorch, try running your code without wandb and see if the error persists.
2. Additionally, can you try the given snippet?

```
import torch
x = torch.rand(5, 3)
print(x)
```

If you code works without wandb and you are able to run the above snippet successfully, then it may be a windows related issue. 

To analyze if this is a windows related error. Provide the windows dump files that get generate after the crash , they are typically found in `C:\Window\Minidump.dmp`, if not, an online search with your specific Windows version should provide a reference.  

Please report back with the above and we will proceed with a review. Thank-you!

**Comment by maxencewynen** [comment]
So I already have a code working with pytorch and pytorch lightning, like previously stated the error only occurs when trying to integrate wandb into my code. I never had any such incidents within the several months I used this PC to do deep learning. 

This might very well be a windows issue, but it is very coincidental that the error really only occurs when running wandb.

Here you'll find a zip to the .dmp file: [out.zip](https://github.com/wandb/client/files/8962236/out.zip)

Many thanks for your help, I am very willing to use wand to enhance my projects

**Comment by MBakirWB** [comment]
Hi @maxencewynen ,

I analyze the dumpfile and the crash is attributed to Nvidia. The crashing module is `nvlddmkm`, a display driver stopped responding, , see attached image.

I did a preliminary search online and got a few hits on addressing this issue, [example](https://www.technewstoday.com/fix-nvlddmkm-sys-error-in-windows/)

![BlueScreenCrash](https://user-images.githubusercontent.com/104453360/176274406-f7e8fc4f-b6b1-4138-8b9b-8c953ee9ad10.PNG)

Can you please attempt to address this locally and retry your script with Wandb. Please provide an update after doing so. 

Thanks

**Comment by MBakirWB** [comment]
HI @maxencewynen , since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!

**Comment by dmitryduev** [comment]
Hey @maxencewynen, many thanks for bringing this to our attention and please accept my apologies for it taking us so long to properly look into. We have updated the vendored version of nvidia-ml-py https://github.com/wandb/wandb/pull/4109 and that PR has been merged into master. It could solve this issue, but I'm not 100% sure it actually will.
Could you please try installing wandb from master and let us know if it works now? Would really appreciate that!

**Comment by djamahl99** [comment]
Having the same issue on a windows machine. Getting a page fault if I include wandb logging in training code that works otherwise. As suggested, have installed master with `pip install git+https://github.com/wandb/wandb`. Blue screen says it's
`PAGE_FAULT_in_NONPAGED_AREA `due to `nvlddmkm.sys`. 

```
Name: wandb
Version: 0.13.3.dev1
Summary: A CLI and library for interacting with the Weights and Biases API.
Home-page: https://github.com/wandb/wandb
Author: Weights & Biases
Author-email: support@wandb.com
License: MIT license
Location: c:\users\djama\anaconda3\envs\pytorch\lib\site-packages
Requires: pathtools, setproctitle, docker-pycreds, PyYAML, sentry-sdk, Click, promise, setuptools, psutil, six, shortuuid, protobuf, GitPython, requests
Required-by:
```

**Comment by djamahl99** [comment]
Had same issue but restarted after installing 0.13.3.dev1 from GitHub and it seems to be fixed 👍

**Comment by BramVanroy** [comment]
Running into the same issue when using `transformers` Trainer with wandb. I'm on 0.13.2 which should include the linked PR (https://github.com/wandb/wandb/pull/4109) but that doesn't seem to solve the problem.

**Comment by dmitryduev** [comment]
@BramVanroy https://github.com/wandb/wandb/pull/4109 will be included in 0.13.3, which is scheduled for this week (0.13.2 was a hot bugfix release). In the meantime, could you please try installing from master or (if that doesn't solve it for you) disable stats logging with `wandb.init(settings=wandb.Settings(_disable_stats=True, _disable_meta=True))`?

**Comment by juansebashr** [comment]
Running into the same issue when working with the artifacts API, I'm on 0.14.2 inside a Python venv
